staging/src/k8s.io/apiserver/pkg/storage/etcd3/etcdfeature/feature_support_checker.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcdfeature
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/apiserver/pkg/storage"
+	"k8s.io/klog/v2"
+)
+
+var (
+	// Define these static versions to use for checking version of etcd, issue on kubernetes #123192
+	v3_4_25 = version.MustParseSemantic("3.4.25")
+	v3_5_0  = version.MustParseSemantic("3.5.0")
+	v3_5_8  = version.MustParseSemantic("3.5.8")
+
+	// DefaultFeatureSupportChecker is a shared global FeatureSupportChecker.
+	DefaultFeatureSupportChecker FeatureSupportChecker = newDefaultFeatureSupportChecker()
```

Comment:
Since you can configure a separate etcd cluster per resource (please double check this) shouldn't we maintain a struct per resource (per etcd client) instead of a global variable ?

Reply:
We check supports by endpoint not etcd client, please take a look at `EndpointSupports(ctx context.Context, c client, endpoint string, feature storage.Feature) (bool, error)`